src/trie.js

```diff
@@ -100,8 +100,10 @@ var marexandre;
             // Check if next character exists in children, and if does dive deeper
             if (copy[j + 1]) {
               var exists2 = tmpTrie.children.hasOwnProperty(copy[j + 1].toString());
-              if (tmpTrie.is_end && !exists2) {
+              if (tmpTrie.is_end) {
```

Comment:
If I'm not mistaken, `tmpTrie.is_end` means it has no children. So the code would probably be clearer like this:
```javascript
if (tmpTrie.is_end) {
  end = start + j;
} else if (!tmpTrie.children.hasOwnProperty(copy[j + 1].toString())) {
  break;
}
```